pkg/trace/config/apply_test.go

```diff
@@ -59,3 +60,119 @@ func TestSplitTag(t *testing.T) {
 		})
 	}
 }
+
+func TestTelemetryEndpointsConfig(t *testing.T) {
+	t.Run("default", func(t *testing.T) {
+		cfg := New()
+		err := cfg.applyDatadogConfig()
+
+		assert := assert.New(t)
+		assert.NoError(err)
+		assert.True(cfg.TelemetryConfig.Enabled)
+		assert.Len(cfg.TelemetryConfig.Endpoints, 1)
+		assert.Equal("https://instrumentation-telemetry-intake.datadoghq.com", cfg.TelemetryConfig.Endpoints[0].Host)
+	})
+
+	t.Run("dd_url", func(t *testing.T) {
+		defer cleanConfig()
+		config.Datadog.Set("apm_config.telemetry.dd_url", "http://example.com/")
+
+		cfg := New()
+		err := cfg.applyDatadogConfig()
+
+		assert := assert.New(t)
+		assert.NoError(err)
+		assert.True(cfg.TelemetryConfig.Enabled)
+		assert.Equal("http://example.com/", cfg.TelemetryConfig.Endpoints[0].Host)
+	})
+
+	t.Run("dd_url-malformed", func(t *testing.T) {
+		defer cleanConfig()
+		config.Datadog.Set("apm_config.telemetry.dd_url", "111://abc.com")
+
+		cfg := New()
+		err := cfg.applyDatadogConfig()
+
+		assert := assert.New(t)
+		assert.NoError(err)
+		assert.True(cfg.TelemetryConfig.Enabled)
+		assert.Equal(cfg.TelemetryConfig.Endpoints[0].Host, "111://abc.com")
+	})
+
+	t.Run("site", func(t *testing.T) {
+		defer cleanConfig()
+		config.Datadog.Set("site", "new_site.example.com")
+
+		cfg := New()
+		err := cfg.applyDatadogConfig()
+		assert := assert.New(t)
+		assert.NoError(err)
+		assert.True(cfg.TelemetryConfig.Enabled)
+		assert.Len(cfg.TelemetryConfig.Endpoints, 1)
+		assert.Equal("https://instrumentation-telemetry-intake.new_site.example.com", cfg.TelemetryConfig.Endpoints[0].Host)
+	})
+
+	t.Run("additional-hosts", func(t *testing.T) {
+		defer cleanConfig()
+		additionalEndpoints := make(map[string]string)
+		additionalEndpoints["test_backend_2.example.com"] = "test_apikey_2"
+		additionalEndpoints["test_backend_3.example.com"] = "test_apikey_3"
```

Comment:
Can you please define the map inline (as suggested in previous comment) in all the tests?